src/services/facilityService.ts

```diff
@@ -19,36 +17,42 @@ export const getFacilityById = async (id: string) : Promise<Facility | null> =>
     return convertedEntity
 }
 
-export const addFacility = async (input: FacilityInput) : Promise<Facility> => {
-    const db = getFirestore()
+export async function addFacility(input: gqlTypes.Facility) {
+    const db = firebase.getFirestore()
    
-    const facilityRef = db.collection('facilities')
-
-    const healthcareProfessionalIds = await mapAndValidateHealthcareProInput(
-        input.healthcareProfessionals as HealthcareProfessionalInput[]
-    )
+    const facilityRef = db.collection('facilities').doc()
+    const healthcareProfessionalRef = db.collection('healthcareProfessionals').doc()
```

Comment:
should move this logic into the healthcareProfessionalService. 

Likely should make a method `addHealthcareProfessionals` that can take an array of them, validate them, and save them.

Reply:
@ermish I tried that initially and couldn't get it to return the healthcareProfessional Ids instead of a promise even with async/await.